grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy

```diff
@@ -157,6 +158,23 @@ class SubmissionController extends AuthenticatedController {
         render (['message': 'document deleted'] as JSON)
     }
 
+    def validateSubmissionNumber() {
+        String projectKey = params?.projectKey
+        String type = params?.type
+        String number = params?.number
+
+        if (!projectKey && !type && !number) {
+            handleIllegalArgumentException(new IllegalArgumentException("invalid params"))
+        }
+
+        Collection<Integer> numbers = queryService.getSubmissionNumber(projectKey, type, number)
+        response.status = 200
+        if (numbers.size() > 0) {
+            render(['valid': false] as JSON)
+        }
+        render(['valid': true] as JSON)
+    }
```

Comment:
Post-merge comment here in case we ever return to this code. There is a fine line between applying business logic in different layers. In this case, we could be a lot more flexible in the UI if we just returned the collection of submission numbers (and perhaps a map of `type` -> list of `number`) and let the UI decide if the number is valid or not. Another approach might be to add this info to the model for project - that would give the UI more information in a single API call.